### ISSUE TITLE
the plant list does not contain referred to taxon

### ISSUE THREAD

**Issue by mfrasca** [Ghini/ghini.desktop#150]
reported by @fruit-github: (#149)

> I have had an issue adding a species more recently that I think should have given a result from the green button. Adding Gymnocalycium damsii, reponse is..
No match found on ThePlantList.org
yet searching ThePlantList.org gives http://www.theplantlist.org/tpl1.1/search?q=Gymnocalycium+damsii

this is caused by the fact that tpl does not contain a taxon that is used as accepted name for the given query.

quick solution to this case:

    @@ -105,7 +105,15 @@ class AskTPL(threading.Thread):
                 else:
                     raise NoResult
                 if found['Accepted ID']:
    -                accepted = ask_tpl(found['Accepted ID'])[0]
    +                logger.debug("found this: %s", str(found))
    +                accepted = ask_tpl(found['Accepted ID'])
    +                logger.debug("ask_tpl on the Accepted ID returns %s", accepted)
    +                if accepted:
    +                    accepted = accepted[0]
    +                else:
    +                    logger.warning("taxon is marked as synonym, but can't "
    +                                   "access accepted form %s", found['Accepted ID'])
    +                    accepted = None
                     logger.debug("%s after second query", self.name)
                 if self.stopped():
                     raise ShouldStopNow('after second query')

**Comment by mfrasca** [comment]
looks like I made an incorrect statement.
the referred to taxon is in TPL but it is at rank subspecies, and my `ask_tpl` method filters away anything that is at rank lower than species, so it does not find it.
_Gymnocalycium damsii_ (K.Schum.) Britton & Rose is a synonym of _Gymnocalycium anisitsii_ subsp. _damsii_ (K.Schum.) G.J.Charles 

to correct the behaviour of ghini, we need adding the handling of infraspecific taxonomic information.

**Comment by mfrasca** [comment]
a quick and incomplete correction:

when you click on the `sp_species_button` button, you immediately get to see a message box informing you that ghini is looking up your data on TPL. (I find it confusing, this lack of feedback).

as soon as the lookup ends (or times out), ghini removes the previous message box and shows the results.
result can be:
* no result
* this is the closest match

and can be followed by:
* closest match is a synonym of this other species
* closest match is a synonym of something at infraspecific rank, which I cannot handle.

**Comment by mfrasca** [comment]
![screenshot from 2016-08-01 11 14 54](https://cloud.githubusercontent.com/assets/122869/17289401/dcdfca96-57d9-11e6-854c-45b62675f487.png)

**Comment by fruit-github** [comment]
That looks a great improvement to me, thanks

I guess from your '**but I'm afraid this should go to 1.0-dev**' that this is not available from a '-u' update - and that itself is not a problem as I see it

**Comment by mfrasca** [comment]
indeed, not yet published on the ghini-1.0 branch, you either wait until I do so, it will be 1.0.60, or you explicitly move to the ghini-1.0-dev branch.

**Comment by fruit-github** [comment]
Happy to wait thanks
